pkgs/development/libraries/flutter-engine/package.nix

```diff
@@ -0,0 +1,417 @@
+
+                cd $out
+                # Generate initial dart package config
+                mkdir -p $out/src/third_party/dart/.dart_tool
+                cp -r --no-preserve=ownership,mode $dartPackageConfig $out/src/third_party/dart/.dart_tool/package_config.json
+
+                # Wrap clang
+                wrapProgram $out/src/buildtools/linux-x64/clang/bin/clang \
+                  --add-flags "-isysroot ${toolchain} ${toolchain.CFLAGS}"
+
+                wrapProgram $out/src/buildtools/linux-x64/clang/bin/ld.lld \
+                  --add-flags "-sysroot ${toolchain} ${toolchain.LDFLAGS}"
+
+                # Run gclient's hooks
+                python3 $out/src/third_party/dart/tools/generate_package_config.py
+                python3 $out/src/third_party/dart/tools/generate_sdk_version_file.py
+                python3 $out/src/tools/remove_stale_pyc_files.py src/tools
+                python3 $out/src/flutter/tools/pub_get_offline.py
+                ${optionalString (targetOS == "wasm") "python3 src/flutter/tools/activate_emsdk.py"}
+              '';
+            };
+          in
+          stdenvNoCC.mkDerivation rec {
+            pname = "flutter-engine-${runtimeMode}";
+            inherit (args') version runtimeMode configureFlags enableParallelBuilding;
+            inherit buildInputs src toolchain;
+
+            nativeBuildInputs = [ git python3 ninja stdenv.cc.libc ];
+
+            extraTargets = (args'.extraTargets or [ ])
+              ++ (optional buildGtk "flutter_gtk");
+
+            passthru = {
+              inherit buildGtk enableFontconfig;
+            };
+
+            requiredSystemFeatures = optional args'.enableParallelBuilding "big-parallel";
+
+            configurePhase = ''
+              # Update time
+              find src -type f -print0 | xargs -0 touch
+
+              # Generate Ninja
+              export PKG_CONFIG_PATH=$toolchain/usr/lib/pkgconfig
+              export PKG_CONFIG_FOR_TARGET=$toolchain/usr/bin/pkg-config
+              export PKG_CONFIG_PATH_FOR_TARGET=$toolchain/usr/lib/pkgconfig
+              export PATH=$toolchain/usr/bin:$PATH
+
+              ./src/flutter/tools/gn \
+                --runtime-mode $runtimeMode \
+                --no-dart-version-git-info \
+                --no-prebuilt-dart-sdk \
+                --no-goma \
+                --clang \
+                --embedder-for-target \
+                --out-dir $out/lib/flutter \
+                --target-sysroot $toolchain \
+                --target-triple ${targetPlatform.config} \
+                --target-dir $runtimeMode \
+                --target-os ${targetOS} \
+                ${optionalString args'.enableFontconfig "--enable-fontconfig"} \
+                ${optionalString (targetOS != "wasm") "--${targetOS}-cpu ${targetCPU}"} \
+                ''${configureFlags[@]}
+            '';
+
+            buildPhase = ''
```

Comment:
> Haven't found a solution to fixing the missing linked libraries
> 
> ```
> $ ldd result/lib/flutter/out/debug/libflutter_engine.so
> 	linux-vdso.so.1 (0x00007ffc45f23000)
> 	libm.so.6 => /nix/store/9xfad3b5z4y00mzmk2wnn4900q0qmxns-glibc-2.35-224/lib/libm.so.6 (0x00007fec9fdab000)
> 	libc.so.6 => /nix/store/9xfad3b5z4y00mzmk2wnn4900q0qmxns-glibc-2.35-224/lib/libc.so.6 (0x00007fec9fba2000)
> 	/nix/store/9xfad3b5z4y00mzmk2wnn4900q0qmxns-glibc-2.35-224/lib64/ld-linux-x86-64.so.2 (0x00007feca2136000)
> $ ldd result/lib/flutter/out/debug/libflutter_linux_gtk.so
> 	linux-vdso.so.1 (0x00007ffc2c3e6000)
> 	libgtk-3.so.0 => not found
> 	libgdk-3.so.0 => not found
> 	libatk-1.0.so.0 => not found
> 	libgio-2.0.so.0 => not found
> 	libgobject-2.0.so.0 => not found
> 	libglib-2.0.so.0 => not found
> 	libepoxy.so.0 => not found
> 	libm.so.6 => /nix/store/9xfad3b5z4y00mzmk2wnn4900q0qmxns-glibc-2.35-224/lib/libm.so.6 (0x00007f15dcc2c000)
> 	libc.so.6 => /nix/store/9xfad3b5z4y00mzmk2wnn4900q0qmxns-glibc-2.35-224/lib/libc.so.6 (0x00007f15dca23000)
> 	/nix/store/9xfad3b5z4y00mzmk2wnn4900q0qmxns-glibc-2.35-224/lib64/ld-linux-x86-64.so.2 (0x00007f15deffc000)
> ```


```
Building flutter_gtk
[1/108] STAMP obj/flutter/shell/platform/embedder/embedder_as_internal_library.inputdeps.stamp
[2/108] CXX obj/flutter/shell/platform/linux/flutter_linux_sources.fl_binary_codec.o
[3/108] CXX obj/flutter/shell/platform/linux/flutter_linux_sources.fl_message_codec.o
...
[103/108] CXX obj/flutter/shell/platform/embedder/embedder_as_internal_library.embedder.o
[104/108] STAMP obj/flutter/shell/platform/embedder/embedder_as_internal_library.stamp
[105/108] STAMP obj/flutter/shell/platform/linux/flutter_linux.stamp
[106/108] SOLINK ./libflutter_linux_gtk.so
[107/108] ACTION //flutter/shell/platform/linux:flutter_gtk(//build/toolchain/linux:clang_x64)
[108/108] STAMP obj/flutter/shell/platform/linux/flutter_gtk.stamp
```

debugging with `ninjaFlags = "-v";`

> [106/108] ../../../../../../../build/flutter-engine-src-3.3.7/src/buildtools/linux-x64/clang/bin/clang++ -shared -Wl,-rpath,\$ORIGIN -Wl,--fatal-warnings -m64 -fPIC -Wl,-z,noexecstack -Wl,-z,now -Wl,-z,relro -Wl,-z,defs -pthread --sysroot=/nix/store/2xzzjvz2vmyqwksnhpkbf2barb9vh3ck-flutter-engine-toolchain-3.3.7  -Wl,-O2 -Wl,--gc-sections -Wl,--as-needed -L/nix/store/2xzzjvz2vmyqwksnhpkbf2barb9vh3ck-flutter-engine-toolchain-3.3.7/nix/store/y1aclp0z9mqpbw206d4ljvwmmq28068g-gtk+3-3.24.35/lib -L/nix/store/2xzzjvz2vmyqwksnhpkbf2barb9vh3ck-flutter-engine-toolchain-3.3.7/nix/store/x96fmh5ydff1yh2x3sl44dd6bwp0ncap-pango-1.50.12/lib -L/nix/store/2xzzjvz2vmyqwksnhpkbf2barb9vh3ck-flutter-engine-toolchain-3.3.7/nix/store/l9n8ads724d5ql0fya9vrys3wq79zcys-harfbuzz-6.0.0/lib -L/nix/store/2xzzjvz2vmyqwksnhpkbf2barb9vh3ck-flutter-engine-toolchain-3.3.7/nix/store/qwn7q1scja4kbgdfrdjxyccz7ll03ig5-at-spi2-core-2.46.0/lib -L/nix/store/2xzzjvz2vmyqwksnhpkbf2barb9vh3ck-flutter-engine-toolchain-3.3.7/nix/store/rqa3gvj23srnjmiydwhwqw9z7qk1rjlq-cairo-1.16.0/lib -L/nix/store/2xzzjvz2vmyqwksnhpkbf2barb9vh3ck-flutter-engine-toolchain-3.3.7/nix/store/7ncncvnr864iangwbvbgbanx1r6wpf79-gdk-pixbuf-2.42.10/lib -L/nix/store/2xzzjvz2vmyqwksnhpkbf2barb9vh3ck-flutter-engine-toolchain-3.3.7/nix/store/i4dqcpppyyq5yqcvw95mv5s11yfyy8pf-glib-2.74.3/lib -L/nix/store/2xzzjvz2vmyqwksnhpkbf2barb9vh3ck-flutter-engine-toolchain-3.3.7/nix/store/pnvm87x9h3qg17njdfh4izbjrl1clpjb-libepoxy-1.5.10/lib  -o ./libflutter_linux_gtk.so -Wl,--build-id=sha1 -Wl,-soname=libflutter_linux_gtk.so @./libflutter_linux_gtk.so.rsp && { ../../../../../../../build/flutter-engine-src-3.3.7/src/buildtools/linux-x64/clang/bin/llvm-readelf -d ./libflutter_linux_gtk.so | grep SONAME ; ../../../../../../../build/flutter-engine-src-3.3.7/src/buildtools/linux-x64/clang/bin/llvm-nm -gD -f posix ./libflutter_linux_gtk.so | cut -f1-2 -d' '; } > ./libflutter_linux_gtk.so.tmp && if ! cmp -s ./libflutter_linux_gtk.so.tmp ./libflutter_linux_gtk.so.TOC; then mv ./libflutter_linux_gtk.so.tmp ./libflutter_linux_gtk.so.TOC; fi && mkdir -p ./lib.unstripped && cp ./libflutter_linux_gtk.so ./lib.unstripped/ && ../../../../../../../build/flutter-engine-src-3.3.7/src/buildtools/linux-x64/clang/bin/llvm-objcopy --strip-all ./lib.unstripped/./libflutter_linux_gtk.so ./libflutter_linux_gtk.so

libgtk-3.so.0 should be found in

```
ls /nix/store/2xzzjvz2vmyqwksnhpkbf2barb9vh3ck-flutter-engine-toolchain-3.3.7/nix/store/y1aclp0z9mqpbw206d4ljvwmmq28068g-gtk+3-3.24.35/lib
girepository-1.0    libgailutil-3.so.0.0.0  libgtk-3.so
gtk-3.0             libgdk-3.so             libgtk-3.so.0
libgailutil-3.so    libgdk-3.so.0           libgtk-3.so.0.2404.31
libgailutil-3.so.0  libgdk-3.so.0.2404.31
```

so ... linker-flags are missing?
should be `clang++ -lgtk-3 -lgdk-3` etc

Reply:
Hmm yeah, that's very strange. Gonna look to see if this is a known issue.

Reply:
Tried that and it does nothing.